tests/foreman/ui/test_webhook.py

```diff
@@ -0,0 +1,83 @@
+"""Test class for Webhook UI
+
+:Requirement: Webhooks
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: Webhooks
+
+:Assignee: gsulliva
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from random import choice
+
+import pytest
+from fauxfactory import gen_string
+from fauxfactory import gen_url
+
+
+@pytest.mark.tier1
+def test_positive_end_to_end(session):
+    """Perform end to end testing for webhooks.
+
+    :id: 9d2072e7-c6b6-4ecb-ac84-e73f4cf76fe4
+
+    :expectedresults: All expected CRUD actions finished successfully
+
+    :CaseImportance: Critical
+    """
+    hook_name = gen_string('alpha')
+    subscribe_to = 'Host Created'
+    target_url = gen_url(subdomain=gen_string('alpha'), scheme=choice(['http', 'https']))
```

Comment:
How long does this test take to execute? I'd rather parametrize this instead of potentially making the test flaky. There's also no guarantee below that the new target url will be using a different scheme than the original - if that's even intended coverage on the update?

Reply:
It takes 4 minutes to execute. Can you define flaky? Also the intention with the url scheme is that webhooks can not be `ftp`, so I'm just ensuring it generates a valid scheme for the test.

Reply:
Flaky would come from a potential bug associated with use of one type of protocol. So when the test randomly chooses the working protocol, things are fine - but when it picks the broken protocol, randomly, the test fails. You go to run it locally to reproduce the CI failure - and it passes because it randomly chose the working protocol.

It may sound far fetched, but @jaryn just unraveled this exact situation in the bookmark tests.

Reply:
Correct, or just consistently use one.  Since the host isn't real, I don't think there's much coverage provided by using both protocols.

If you want to validate the allowed protocol types, given the execution time, I'd consider having a separate test function to fill the UI field with http/https/ftp/file protocols and check the validation behavior specifically.

Reply:
Agreed. Going to limit to just `http`, since we aren't gaining much coverage.